Stack/Opc.Ua.Core/Security/Certificates/X509Utils.cs

```diff
@@ -466,6 +491,50 @@ public static List<string> ParseDistinguishedName(string name)
             return fields;
         }
 
+        /// <summary>
+        /// Return if a certificate has a ECDsa signature.
+        /// </summary>
+        /// <param name="cert">The certificate to test.</param>
+        public static bool IsECDsaSignature(X509Certificate2 cert)
+        {
+            return X509PfxUtils.IsECDsaSignature(cert);
+        }
+
+        /// <summary>
+        /// Return a qualifier string if a ECDsa signature algorithm used.
+        /// </summary>
+        /// <param name="certificate">The certificate.</param>
+        public static string GetECDsaQualifier(X509Certificate2 certificate)
+        {
+            return EccUtils.GetECDsaQualifier(certificate);
+        }
+
+        /// <summary>
+        /// Verify RSA key pair of two certificates.
+        /// </summary>
+        public static bool VerifyKeyPair(
+            X509Certificate2 certWithPublicKey,
+            X509Certificate2 certWithPrivateKey,
+            bool throwOnError = false)
+        {
+            return X509PfxUtils.VerifyKeyPair(certWithPublicKey, certWithPrivateKey, throwOnError);
+        }
+
+        /// <summary>
+        /// Verify ECDsa key pair of two certificates.
+        /// </summary>
+        public static bool VerifyECDsaKeyPair(
```

Comment:
imho we should try to deprecate RSA Specific functions where possible and also try to avoid ecc specific functions where possible and insteady provide Functions supporting both certificate types wherever possible